From df5345946a10d9b958e1d0249a0df9bc7467efec Mon Sep 17 00:00:00 2001
From: Andrey Gelman <andreyg@compulab.co.il>
Date: Wed, 13 Mar 2013 14:13:29 +0200
Subject: wm8731: fix the list of reported supported sampling rates

Prior to this fix, wm8731 was reporting some sampling rates
it was not able to play back, resulting in players sending
unsupported sampling rates for playback.
---
 sound/soc/codecs/wm8731.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/sound/soc/codecs/wm8731.c b/sound/soc/codecs/wm8731.c
index f5a0ec4..a17f154 100644
--- a/sound/soc/codecs/wm8731.c
+++ b/sound/soc/codecs/wm8731.c
@@ -282,6 +282,9 @@ static inline int get_coeff(int mclk, int rate)
 		if (coeff_div[i].rate == rate && coeff_div[i].mclk == mclk)
 			return i;
 	}
+
+	pr_info("wm8731: inappropriate clock rates: mclk = %d  sampling = %d \n",
+			mclk, rate);
 	return 0;
 }
 
@@ -470,7 +473,9 @@ static int wm8731_set_bias_level(struct snd_soc_codec *codec,
 	return 0;
 }
 
-#define WM8731_RATES SNDRV_PCM_RATE_8000_96000
+#define WM8731_RATES	(SNDRV_PCM_RATE_8000 | SNDRV_PCM_RATE_32000 | \
+			SNDRV_PCM_RATE_44100 | SNDRV_PCM_RATE_48000 | \
+			SNDRV_PCM_RATE_88200 | SNDRV_PCM_RATE_96000)
 
 #define WM8731_FORMATS (SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S20_3LE |\
 	SNDRV_PCM_FMTBIT_S24_LE)
-- 
1.7.9.5

